networkx/algorithms/centrality/domirank.py

```diff
@@ -0,0 +1,303 @@
+
+    converged: boolean
+        Whether the centrality computation converged.
+
+    Examples
+    --------
+    >>> G = nx.path_graph(5)
+    >>> centrality, sigma, converged = nx.domirank(G)
+    >>> print([f"{node} {centrality[node]:0.2f}" for node in centrality])
+    ['0 -0.54', '1 1.98', '2 -1.08', '3 1.98', '4 -0.54']
+
+    Raises
+    ------
+    NetworkXPointlessConcept
+        If the graph `G` is the null graph.
+
+    NetworkXUnfeasible
+        If `alpha` is negative (and thus outside its bounds): ``alpha < 0``.
+
+    NetworkXUnfeasible
+        If ``alpha > 1`` with the ``analytical=False`` argument.
+
+    Warning
+        If supercharging the competition parameter for the analytical solution: ``alpha > 1``.
+
+    Warning
+        If one is using the analytical solution for a large system, i.e. ``len(G) > 5000``, as the algorithm will be slow.
+
+    See Also
+    --------
+    :func:`~networkx.algorithms.centrality.degree_centrality`
+    :func:`~networkx.algorithms.link_analysis.pagerank_alg.pagerank`
+    :func:`~networkx.algorithms.link_analysis.hits_alg.hits`
+
+    References
+    ----------
+    .. [1] Engsig, M., Tejedor, A., Moreno, Y. et al.
+        "DomiRank Centrality reveals structural fragility of complex networks via node dominance."
+        Nat Commun 15, 56 (2024). https://doi.org/10.1038/s41467-023-44257-0
+    """
+    import numpy as np
+    import scipy as sp
+
+    if len(G) == 0:
+        raise nx.NetworkXPointlessConcept(
+            "cannot compute centrality for the null graph"
+        )
+    GAdj = nx.to_scipy_sparse_array(G)  # convert to scipy sparse csr array
+
+    # Here we renormalize alpha with the smallest eigenvalue (most negative eigenvalue) by calling the "hidden" function _find_smallest_eigenvalue()
+    # Note, this function always uses the recursive definition
+    if analytical == False:
+        # If the recursive formulation is used, the alpha has to be bounded (competition parameter).
+        if alpha < 0:
+            raise nx.NetworkXUnfeasible(
+                "the competition parameter alpha must be positive and bounded such that alpha: [0,1]"
+            )
+        if alpha > 1:
+            raise nx.NetworkXUnfeasible(
+                "supercharging the competition parameter (alpha > 1) requires the <analytical = True> flag"
+            )
+        sigma = np.abs(
+            alpha
+            / _find_smallest_eigenvalue(
+                GAdj,
+                maxDepth=int(max_iter / 5),
+                dt=dt,
+                epsilon=epsilon,
+                max_iter=max_iter,
+                patience=patience,
+            )
+        )
+        # store this to prevent more redundant calculations in the future
+        pGAdj = sigma * GAdj.astype(np.float32)
+        # initalize a proportionally (to system size) small non-zero uniform vector
+        Psi = np.ones(pGAdj.shape[0]).astype(np.float32) / pGAdj.shape[0]
```

Comment:
```suggestion
        psi = np.ones(pGAdj.shape[0], dtype=np.float32) / pGAdj.shape[0]
```

I personally also prefer lowercase variable names (here and in some other places), but feel free to disregard.